src/Microsoft.OData.ModelBuilder/Vocabularies/VocabularyTermConfiguration.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Vocabularies;
+
+namespace Microsoft.OData.ModelBuilder
+{
+    /// <summary>
+    /// Base vocabulary configuration.
+    /// </summary>
+    public abstract partial class VocabularyTermConfiguration : IRecord
+    {
+        private readonly string _termName;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="VocabularyTermConfiguration"/>
+        /// </summary>
+        /// <param name="termName">The name of the <see cref="IEdmTerm"/> being built.</param>
+        public VocabularyTermConfiguration(string termName) => _termName = termName;
+
+        /// <inheritdoc/>
+        public abstract IEdmExpression ToEdmExpression();
+
+        /// <summary>
+        /// Sets the vocabulary annotation on the model's target.
+        /// </summary>
+        /// <param name="model">The <see cref="IEdmModel"/> having reference vocabulary models.</param>
+        /// <param name="target">The <see cref="IEdmVocabularyAnnotatable"/> to set annotations on.</param>
+        public virtual void SetVocabularyAnnotations(EdmModel model, IEdmVocabularyAnnotatable target)
+        {
+            _ = model ?? throw new ArgumentNullException(nameof(model));
+            _ = target ?? throw new ArgumentNullException(nameof(target));
```

Comment:
Maybe we can use 

if (model != null)
{  
    throw new .....;
}

in this matter, it only one test without the assignment operation.

Reply:
This is the [C#7 throw expression](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/throw#the-throw-expression) and we want to throw if the model **is** null...